editor/editor_settings.cpp

```diff
@@ -1122,8 +1122,8 @@ Variant _EDITOR_DEF(const String &p_setting, const Variant &p_default, bool p_re
 		ret = EDITOR_GET(p_setting);
 	} else {
 		EditorSettings::get_singleton()->set_manually(p_setting, p_default);
-		EditorSettings::get_singleton()->set_restart_if_changed(p_setting, p_restart_if_changed);
 	}
+	EditorSettings::get_singleton()->set_restart_if_changed(p_setting, p_restart_if_changed);
```

Comment:
Is this required?

Reply:
I believe so, as there is a editor setting cache mechanism, we have to set the restart_if_changed if the value is directly read from `EDITOR_GET`.